common/app/common/dfp/TagSponsorship.scala

```diff
@@ -100,10 +83,18 @@ case class HighMerchandisingLineItem(
   ) {
   val customTargets = customTargetSet.map(_.targets)
   val editions = customTargets.flatMap(sequence => sequence.filter((target) => target.name == "edition")).map(target => target.values)
-}
 
-case class HighMerchandisingLineItems(items: Seq[HighMerchandisingLineItem]) {
-  val sortedItems = items.sortBy(_.name)
+
+  def matchesAdUnitAndTag (adUnitSuffix: String, pageTags:Seq[Tag]): Boolean = {
+
+    def tagNames = pageTags map (_.name) map (_.replaceAll(" ","-").toLowerCase)
+
+    def matchesTag: Boolean = tagNames.exists(tags.contains)
+
+    def matchesAdUnit: Boolean = adUnits.exists(_.path contains adUnitSuffix)
```

Comment:
Good to know - I thought that since these methods will only be called once (and so the cost of calling them will be the same) that, using a `def`, there will be no subsequent val-storage cost. I suppose given the piddly amount of space we are talking, it makes more sense to be clearer about intent rather than worrying about a few bytes.